addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4258,6 +4258,10 @@ registry.BackgroundPosition = SnippetOptionWidget.extend({
             left: position[0] / 100 * delta.x || 0,
             top: position[1] / 100 * delta.y || 0,
         };
+        // Make sure element is in visible area.
+        this.$target[0].scrollIntoView(false);
+        // And wait for new layout offsets to be computed before positioning buttons.
+        await new Promise(resolve => setTimeout(resolve));
```

Comment:
@bso-odoo I tested this flow on Firefox and it does not work:

- Select a snippet
- Scroll so that it is half visible in the viewport
- Hit the bg position button
=> The page scrolls to make the snippet fully visible but the buttons to apply/discard are not there

Reply:
Very strange... pressing "escape" redraws them in the right position.

Reply:
To cover the partial hiding, an offset was computed to position the buttons, but since we now scroll to show the block, the buttons can remain at `top: 0` defined in their template.